lib/puppet/settings.rb

```diff
@@ -385,6 +385,19 @@ def initialize_app_defaults(app_defaults)
     call_hooks_deferred_to_application_initialization
     issue_deprecations
 
+    run_mode = Puppet::Util::RunMode[self.preferred_run_mode]
+    if run_mode.agent? || run_mode.server?
+      if self.set_in_section?(:masterport, run_mode.name) && !self.set_in_section?(:serverport, run_mode.name)
+        self[:serverport] = self[:masterport]
```

Comment:
notes to future self:

if `:masterport` is set in `[agent]` and `:serverport` isn't, use the former.